src/main/java/net/openhft/chronicle/threads/internal/ThreadMonitorHarness.java

```diff
@@ -37,16 +48,19 @@ public boolean action() throws InvalidEventHandlerException {
             throw new InvalidEventHandlerException();
         }
         long startedNS = thread.startedNS();
-        long nowNS = System.nanoTime();
+        long nowNS = timeSupplier.getAsLong();
 
         // Record lastActionCall time on every call to prevent false-positive "monitorThreadDelayed" reports
         long actionCallDelay = nowNS - this.lastActionCall;
         this.lastActionCall = nowNS;
 
-        if (startedNS == 0 || startedNS == Long.MAX_VALUE) {
-            thread.resetTimers();
+        if (startedNS == 0 || startedNS == NOT_IN_A_LOOP) {
             return false;
         }
+        if (startedNS != lastStartedNS) {
+            thread.resetTimers();
+            lastStartedNS = startedNS;
+        }
```

Comment:
This is the "meat" of the change. We now just bail when the event loop indicates it's not in an iteration, and we reset the timers any time that we notice the `startedNS` has changed.